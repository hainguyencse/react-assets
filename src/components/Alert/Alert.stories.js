import React from 'react';
import { action } from '@storybook/addon-actions';
import Alert from '.';

export default [
  {
    title: 'Success',
    component: <Alert title="Success" type="success" content="This is a success alert" onDismiss={action('dismiss')} />
  },
  {
    title: 'Warning',
    component: <Alert title="Warning" type="warning" content="This is a warning alert" onDismiss={action('dismiss')} />
  },
  {
    title: 'Danger',
    component: <Alert title="Danger" type="danger" content="This is a danger alert" onDismiss={action('dismiss')} />
  },
];
