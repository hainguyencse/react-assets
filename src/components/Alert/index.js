import React from 'react';
import PropTypes from 'prop-types';

/**
 * Alert show important content to user
 */
const Alert = ({
  title, content, type, onDismiss
}) => {
  let iconName;
  const renderIcon = () => {
    switch (type) {
      case 'warning': {
        iconName = 'fa-warning';
        break;
      }
      case 'danger': {
        iconName = 'fa-ban';
        break;
      }
      default: {
        iconName = 'fa-check';
      }
    }

    return (
      <i className={`icon fa ${iconName}`} />
    );
  };

  return (
    <div className={`alert alert-${type} alert-dismissible`}>
      <button type="button" className="close" onClick={onDismiss}>Ã—</button>
      <h4>
        {renderIcon()}
        {' '}
        {title}
      </h4>
      {content}
    </div>
  );
};

Alert.propTypes = {
  /**
   * Title of Alert
   */
  title: PropTypes.string,
  /**
   * Content displayed in the Alert body
   */
  content: PropTypes.string,
  /**
   * Type of alert
   */
  type: PropTypes.oneOf(['success', 'warning', 'danger']),
  /**
   * Function callback when alert is dismissed
   */
  onDismiss: PropTypes.func,
};

Alert.defaultProps = {
  title: 'Alert',
  content: 'Alert content',
  type: 'success',
  onDismiss: () => {},
};

export default Alert;
