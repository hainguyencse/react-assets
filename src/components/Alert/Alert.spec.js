import React from 'react';
import Alert from '.';

describe('Alert', () => {
  test('should render properly', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<Alert title="Alert" type="success" content="Success"/>);

    expect(wrapper).toMatchSnapshot();
  });
});
