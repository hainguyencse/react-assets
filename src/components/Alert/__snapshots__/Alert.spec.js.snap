// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Alert
    content="Success"
    onDismiss={[Function]}
    title="Alert"
    type="success"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="close"
          onClick={[Function]}
          type="button"
        >
          ×
        </button>,
        <h4>
          <i
            className="icon fa fa-check"
          />
           
          Alert
        </h4>,
        "Success",
      ],
      "className": "alert alert-success alert-dismissible",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "×",
          "className": "close",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "×",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="icon fa fa-check"
            />,
            " ",
            "Alert",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "icon fa fa-check",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          " ",
          "Alert",
        ],
        "type": "h4",
      },
      "Success",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="close"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>,
          <h4>
            <i
              className="icon fa fa-check"
            />
             
            Alert
          </h4>,
          "Success",
        ],
        "className": "alert alert-success alert-dismissible",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "×",
            "className": "close",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "×",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="icon fa fa-check"
              />,
              " ",
              "Alert",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "icon fa fa-check",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " ",
            "Alert",
          ],
          "type": "h4",
        },
        "Success",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
